src/enzyme_ad/jax/Dialect/EnzymeXLAOps.td

```diff
@@ -780,6 +780,276 @@ def GesvjOp : EnzymeXLA_Op<"lapack.gesvj", [Pure]> {
   }];
 }
 
+// Special Functions - Bessel Functions
+
+def BesselJ : EnzymeXLA_Op<"special.besselj", [Pure, SameOperandsAndResultType, Elementwise]> {
+  let summary = "Bessel function of the first kind of order nu at z";
+  
+  let arguments = (ins
+    HLO_Tensor:$nu,
+    HLO_Tensor:$z
+  );
+  
+  let results = (outs
+    HLO_Tensor:$res
+  );
+}
+
+def BesselJ0 : EnzymeXLA_Op<"special.besselj0", [Pure, SameOperandsAndResultType, Elementwise]> {
```

Comment:
Is there a reason for us to specialize j0 here? If we have the same implementation it lowers into anyways, no need for now

Reply:
I imagined those might have custom approximation lowerings but I'm not sure about that. Will see if I can find something

Reply:
for one, mpfr seems to have separate implementations for the different bessels https://github.com/JuliaMath/SpecialFunctions.jl/blob/1743a8b7ac1565213e87de418765c594720929b6/src/bessel.jl#L682C1-L709C4
I'm not sure yet what the lowering in stablehlo would look like.